osquery/core/database/tests/in_memory_database_tests.cpp

```diff
@@ -0,0 +1,66 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <gtest/gtest.h>
+
+#include <osquery/core/database/in_memory_database.h>
+#include <osquery/database.h>
+
+namespace osquery {
+
+GTEST_TEST(InMemoryDatabaseTest, test_open) {
+  auto db = std::make_unique<InMemoryDatabase>("test");
+  auto result = db->open();
+  EXPECT_TRUE(result);
+  db->close();
+}
+
+GTEST_TEST(InMemoryDatabaseTest, test_destroy) {
+  auto db = std::make_unique<InMemoryDatabase>("test");
+  auto result = db->destroyDB();
+  EXPECT_TRUE(result);
+}
+
+GTEST_TEST(InMemoryDatabaseTest, test_put) {
+  auto db = std::make_unique<InMemoryDatabase>("test");
+  auto result = db->putInt32("test", "test", 23);
+  EXPECT_FALSE(result);
+  EXPECT_EQ(result.getError(), DatabaseError::DatabaseIsNotOpen);
+  result = db->open();
+  EXPECT_TRUE(result);
+  EXPECT_TRUE(db->putInt32(kPersistentSettings, "test_key_int", 12));
+  auto int_value = db->getInt32(kPersistentSettings, "test_key_int");
+  EXPECT_TRUE(int_value);
+  EXPECT_EQ(int_value.take(), 12);
+
+  EXPECT_TRUE(db->putString(kPersistentSettings, "test_key_string", "string"));
+  auto string_value = db->getString(kPersistentSettings, "test_key_string");
+  EXPECT_TRUE(string_value);
+  EXPECT_EQ(string_value.take(), "string");
```

Comment:
May I ask you to split this test? At least: put without to non opened db, put int, put string?